components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -386,6 +403,77 @@ void AdsServiceImpl::Start() {
   BackgroundHelper::GetInstance()->AddObserver(this);
 }
 
+void AdsServiceImpl::MaybeShowFirstLaunchNotification() {
+  if (!ShouldShowAdsNotification() ||
+      !profile_->GetPrefs()->GetBoolean(
+          brave_rewards::prefs::kBraveRewardsEnabled))
+    return;
+
+  if (profile_->GetPrefs()->GetTime(
+          prefs::kBraveAdsLaunchNotificationTimestamp) == base::Time()) {
+      profile_->GetPrefs()->SetTime(
+          prefs::kBraveAdsLaunchNotificationTimestamp,
+        base::Time::Now());
+  }
+
+  auto* rewards_service =
+      brave_rewards::RewardsServiceFactory::GetForProfile(profile_);
+
+  auto* rewards_notification_service =
+      rewards_service->GetNotificationService();
+
+  RewardsNotificationService::RewardsNotificationArgs args;
+  rewards_notification_service->AddNotification(
+      RewardsNotificationService::REWARDS_NOTIFICATION_ADS_LAUNCH,
+      args,
+      rewards_notification_ads_launch);
+
+  uint32_t timeout_length = IsTestingEnv()
+      ? (5 * 60) /* 5 minutes */
+      : (24 * 60 * 60 * 7); /* 7 days */
+
+  auto timeout = base::Time::Now() - profile_->GetPrefs()->GetTime(
+      prefs::kBraveAdsLaunchNotificationTimestamp) +
+      base::TimeDelta::FromSeconds(timeout_length);
+
+  uint32_t timer_id = next_timer_id();
+  ads_launch_id_ = timer_id;
+  timers_[timer_id] = std::make_unique<base::OneShotTimer>();
+  timers_[timer_id]->Start(FROM_HERE,
+      timeout,
+      base::BindOnce(&AdsServiceImpl::FirstLaunchNotificationTimedOut,
+          AsWeakPtr(),
+          timer_id,
+          rewards_notification_ads_launch));
+}
+
+void AdsServiceImpl::FirstLaunchNotificationTimedOut(uint32_t timer_id,
+                                          const std::string& notification_id) {
+  timers_.erase(timer_id);
+  auto* rewards_service_ =
+      brave_rewards::RewardsServiceFactory::GetForProfile(profile_);
+
+  auto* rewards_notification_service =
+      rewards_service_->GetNotificationService();
+
+  rewards_notification_service->DeleteNotification(notification_id);
+}
+
+void AdsServiceImpl::OnNotificationDeleted(
+      RewardsNotificationService* rewards_notification_service,
+      const RewardsNotificationService::RewardsNotification& notification) {
+  if (notification.id_ == rewards_notification_ads_launch) {
+    profile_->GetPrefs()->SetBoolean(
+      prefs::kBraveAdsShowAdsNotification, false);
+  }
+}
+
+bool AdsServiceImpl::IsTestingEnv() {
+  const base::CommandLine& command_line =
+      *base::CommandLine::ForCurrentProcess();
+  return command_line.HasSwitch(switches::kTesting);
```

Comment:
By using this command-line switch the `Easter Egg` feature will be enabled and potentially other features in the future

Reply:
function removed